backend/src/main/java/codezap/category/dto/request/UpdateAllCategoriesRequest.java

```
@@ -0,0 +1,24 @@
+package codezap.category.dto.request;
+
+import java.util.List;
+
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+
+import codezap.global.validation.ValidationGroups.NotNullGroup;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+public record UpdateAllCategoriesRequest(
```

(⚡어필의 차원에서 zap을 사용합니다 ㅎㅎ )

해당 dto에서 fail fast를 위해 소스 코드 request처럼 SourceCodesOrdinalValidator를 이용해서 dto에서도 순서 검증하는 건 어떨까요?

현재 해당 로직이 템플릿 수정 로직과 매우 유사하다고 생각합니다. 그래서 현재 SourceCodesOrdinalValidator의 이름을 OrdinalValidator로 변경하여 소스코드에서만 사용하지 않고 이렇게 순서 검증이 필요한 곳에서 재사용하면 좋을 것 같아요 ~! 

이렇게 한다면 ValidatedSourceCodesOrdinalRequest의 
`@SourceCodesOrdinal(message = "소스 코드 순서가 잘못되었습니다.", groups = SourceCodeOrdinalGroup.class)`
이 메시지를 `순서가 잘못되었습니다.` 로 어디서든 사용될 수 있게 변경한다면 더 재사용하는데 좋을 것 같아요~

이에 대해 어떻게 생각하시나요?

> 만약 변경에 동의하시다면 지금 당장 클래스 이름은 변경 안하면 좋을 것 같아요 ,,! 해당 클래스를 #991 에서 사용하고 있어서 추후 둘 다 머지되고 이름 변경하면 좋을 것 같아요~

---

이미 동일한 기능을 하는 `Validator`가 존재하니 좋은 의견이네요~~
소스코드와 카테고리의 동작이 언제까지 동일할지 모르니 기존의 클래스를 완전 재사용하는 것이 조금은 조심스럽기도 하지만, 나중에 동작이 달라져야 할 때 상속을 통한 분리를 고려해 봐도 좋을 것 같고요~~